sample-scsi/src/main.rs

```diff
@@ -122,21 +122,23 @@ fn main() {
 		}
 	}
 
-	let mut pages = SCSIPages::new(&dev);
-	if let Ok(supported_pages) = pages.supported_pages() {
-		for p in supported_pages {
-			if p == 00 { continue; }
-
-			print!("=== [{:02x}] {} ===\n", p, page_name(p));
-			match p {
-				// already in cli
-				0x02...0x06 | 0x0d | 0x0e => (),
-				0x10 => print!("{:#?}\n", pages.self_test_results()),
-				0x2f => print!("{:#?}\n", pages.informational_exceptions()),
-				_ => (),
-			}
+	let mut pages = SCSIPages::new(&dev).unwrap();
+	let supported_pages = pages.supported_pages()
+	    .iter()
+	    .cloned()
+	    .collect::<Vec<_>>();
+	for p in supported_pages {
+		if p == 00 { continue; }
+
+		print!("=== [{:02x}] {} ===\n", p, page_name(p));
+		match p {
+			// already in cli
+			0x02...0x06 | 0x0d | 0x0e => (),
+			0x10 => print!("{:#?}\n", pages.self_test_results()),
+			0x2f => print!("{:#?}\n", pages.informational_exceptions()),
+			_ => (),
```

Comment:
Out of curiosity: may I ask what prompted you to do this seemingly irrelevant change?

Reply:
The borrow checker.  It didn't like the old code, because `pages` was borrowed read-only at line 126, and also borrowed read-write at lines 134 & 135 while the read-only borrow was still active.  The new method `collect`s the supported pages so there won't be a borrow conflict.

Reply:
Oh, I see. It seems it's been a looong while since I last tried building this subcrate…